### ISSUE TITLE
Missing documentation

### ISSUE THREAD

**Issue by martinrulf** [LazZiya/XLocalizer.DB.UI#3]
Hi, 

Is there some documentation describing how to make this Work? 
I mean how to add/register this NuGet XLocalizer.DB.UI with existing ASP.NET Core Project? :) 

Btw: documentation for the rest is great and translation service works well but I was not able to find how to setup this part...

Thank you for your help in advance

**Comment by LazZiya** [comment]
Hi @martinrulf ,

Here is the steps required to use `XLocalizer.DB.UI`:

- Install from nuget
````
PM > install-package XLocalizer.DB.UI
````

- In your main project add links to the "Cultures" and "Resources" pages, both are under "XLocalizer" area inside `XLocalizer.DB.UI`:
````html
<a class="nav-link text-dark" asp-area="XLocalizer" asp-page="/Cultures/Index" localize-content>DB Cultures</a>

<a class="nav-link text-dark" asp-area="XLocalizer" asp-page="/Resources/Index" localize-content>DB Resources</a>
````
Here you can find a [sample project](https://github.com/LazZiya/XLocalizer.Samples/tree/master/DbLocalizationSample/DbLocalizationSample) with DB and DB.UI nugets.

> Notice: I will add docs for this nuget soon :)

Kind regards,
Ziya

**Comment by martinrulf** [comment]
Hi @LazZiya,

Thank you for a quick answer.
Currently I am using MVC + Razor Pages together as the project is in fact a microservice with REST and GraphQL APIs

It was a bit tricky to configure it, still the icons from **XLocalizer.DB.UI** are not fetched, and I do not know why, probably the embedded resources were not loaded.

In case of links, I had to use links in this form.
```html
    <p><a href="/XLocalizer/Cultures/Index">Manage XLocalizer Cultures</a></p>
    <p><a href="/XLocalizer/Resources/Index">Manage XLocalizer Resources</a></p>
```
![image](https://user-images.githubusercontent.com/37901240/125105579-e7211a80-e0de-11eb-8b9d-1a8d083420f2.png)
else the links were rendered as a plain text not as links, again I do not know why.

You can see my current result below, you can notice that buttons are rendered only as colored rectangles.
![image](https://user-images.githubusercontent.com/37901240/125105291-9a3d4400-e0de-11eb-8927-70074313c200.png)
![image](https://user-images.githubusercontent.com/37901240/125105325-a3c6ac00-e0de-11eb-9407-537b757a1686.png)

I had to add all these into project
![image](https://user-images.githubusercontent.com/37901240/125105915-4bdc7500-e0df-11eb-9bec-da2a7350df56.png)
![image](https://user-images.githubusercontent.com/37901240/125105959-5c8ceb00-e0df-11eb-9458-f1b28a4dde46.png)
![image](https://user-images.githubusercontent.com/37901240/125105978-631b6280-e0df-11eb-90fc-142eefd196ae.png)

I will dig into it more later.

Anyway thank you very much for provided example it helped a lot as I am backend developer, making WEB (HTML) is not my expertise :)

**Comment by martinrulf** [comment]
Hi @LazZiya,

adding _ViewImports.cshtml into Views
with this content
```cshtml
@using Microsoft.AspNetCore.Identity
@namespace Konzole.WarehouseConnector.Api.Views
@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
@addTagHelper *, LazZiya.TagHelpers
@addTagHelper *, XLocalizer.TagHelpers
```

did the job.
This was missing.
@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers

Ok, only the missing images remain. :)

**Comment by LazZiya** [comment]
Hi @martinrulf ,

I'm not sure if this is related to the folder structure of the project, but It seems you are missing fontawesome, you can find the one I used in [`_Layout.cshtml`](https://github.com/LazZiya/XLocalizer.DB.UI/blob/master/XLocalizer.DB.UI/Areas/XLocalizer/Pages/Shared/_Layout.cshtml) page:

````html
<!-- fontawesome -->
<script src="https://kit.fontawesome.com/272d3a8e61.js" crossorigin="anonymous"></script>

<!-- flag icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/flag-icon-css/3.4.6/css/flag-icon.min.css" />
````

btw, I do fullstack development, but more specialized in backend, frontend is a nightmare sometimes :)

**Comment by martinrulf** [comment]
Hi @LazZiya,

Thank you for the hints :) 
Unfortunately, it seems the problem is with CORS.

If I start your demo and check the responses from the server in chrome debug console I can see this
![image](https://user-images.githubusercontent.com/37901240/125314291-59d80300-e336-11eb-9d14-f2739b10ae33.png)
![image](https://user-images.githubusercontent.com/37901240/125314524-8be96500-e336-11eb-9b27-124b47be48f9.png)
As you can see there was only a HTTP GET method called for getting  https://kit.fontawesome.com/272d3a8e61.js

In my case a preflight (HTTP OPTIONS) is called and it seems that it is not supported by their server
![image](https://user-images.githubusercontent.com/37901240/125314855-dc60c280-e336-11eb-9103-cd75b3107258.png)
![image](https://user-images.githubusercontent.com/37901240/125315104-14680580-e337-11eb-9522-c992fbf0588e.png)

I found something about it here
https://github.com/FortAwesome/Font-Awesome/issues/17402

So I added a comment 
https://github.com/FortAwesome/Font-Awesome/issues/17402#issuecomment-878400046

Anyway, it is more nice to have feature than a must have so I will focus more on the rest of the app. :)

And yes, I fully agree, frontend is a hell in case of web development... I've made many desktop apps in WinForms, WPF, UWP and it was never so frustrating as it is in case of web development... :)

BTW one more question, can is somewhat enable a support for paging for resources page? 
As there are only first 10 records and the rest I am able to get only by filtering those... :)

Thank you for your help it is much appreciated.

**Comment by LazZiya** [comment]
Hi @martinrulf 

I think you can discard the fontawesome script in the nuget and add your own in your _Layout.cshtml.

### Option 1:
````html
<!-- Fontawesome -->
<script defer src="https://use.fontawesome.com/releases/v5.15.1/js/all.js" crossorigin="anonymous"></script>
````

### Option 2:
[Download free version of fontawesome](https://fontawesome.com/v5.15/how-to-use/on-the-web/setup/getting-started) and include it in your project.

### Option 3:
[Provide your email address and obtain a cdn link.](https://fontawesome.com/v4.7/get-started/)

### Option 4:
[Register at fontawesome and create your own kit.](https://fontawesome.com/kits)

---

## Paging support
Do you mean it is not showing the next 10 records when you click on the relevant page number? it should work fine without any additional settings, can you provide me more details?

The project is using another nuget named [LazZiya.TagHelpers](https://github.com/LazZiya/TagHelpers) that I created for paging support :) You can customize the paging control via `appSettings.json`, see the [json settings docs](https://docs.ziyad.info/en/LazZiya.TagHelpers/v5.0/Paging-TagHelper-Json-Settings.md) and [Paging TagHelper Attributes](https://docs.ziyad.info/en/LazZiya.TagHelpers/v5.0/Paging-TagHelper-Attributes.md) for more details.

**Comment by martinrulf** [comment]
Yes it did the trick, eventually I use libman, it works great... :)
![image](https://user-images.githubusercontent.com/37901240/125452393-7e2bbd50-bfd3-4958-8bb8-0dd69a8e6a20.png)
![image](https://user-images.githubusercontent.com/37901240/125452432-8a76c640-bb80-4f1b-b836-20d332233f9d.png)
![image](https://user-images.githubusercontent.com/37901240/125452467-530dae02-3a0c-49b5-b5f4-842d55243bb7.png)
![image](https://user-images.githubusercontent.com/37901240/125452519-1f1a902a-3c21-40d0-9aac-571e18048b85.png)

Regarding the paging, there is not any paging, even when I tried your demo the paging was not there. :)
What you get is this
![image](https://user-images.githubusercontent.com/37901240/125452680-7f5e9fe0-bc26-4e9d-9091-427e721aed96.png)
 I can't see any page selector or page navigation... :)

**Comment by LazZiya** [comment]
Ooops! now that is a bug 🤦🏽‍♂️ I mean the pagination :(
fix is coming within a few minutes...

**Comment by LazZiya** [comment]
Fix provided in v1.0.1 and it is available in nuget.

btw, I also updated the fontawesome script src, hopefuly this one will work without any workaround :)

![image](https://user-images.githubusercontent.com/22244584/125459908-824a8c46-8de5-402e-a573-7ef31e2549db.png)

**Comment by martinrulf** [comment]
Great! It works like a charm... :)

![image](https://user-images.githubusercontent.com/37901240/125463398-ae8b8d88-cb9c-41b8-b23d-057c7e6bcfea.png)

Thank you
